step/matcher/graph_pipeline.go

```diff
@@ -0,0 +1,58 @@
+package matcher
+
+import (
+	"regexp"
+
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-steplib/bitrise-step-pull-intermediate-files/model"
+)
+
+type graphPipelineMatcher struct {
+	finishedWorkflows model.FinishedWorkflows
+	targetNames       []string
+	logger            log.Logger
+}
+
+func newGraphPipelineMatcher(finishedWorkflows model.FinishedWorkflows, targetNames []string, logger log.Logger) graphPipelineMatcher {
+	return graphPipelineMatcher{
+		finishedWorkflows: finishedWorkflows,
+		targetNames:       targetNames,
+		logger:            logger,
+	}
+}
+
+func (g graphPipelineMatcher) Matches() ([]string, error) {
+	var executedWorkflows []model.Workflow
+	for _, workflow := range g.finishedWorkflows {
+		if workflow.ExternalId == "" {
+			g.logger.Printf("Skipping workflow %s becaues it was not executed.", workflow.Name)
```

Comment:
We can consider quoting workflow names (not sure if you have conventions here). I think it results in a more readable output.
```suggestion
			g.logger.Printf("Skipping workflow %q because it was not executed.", workflow.Name)
```